src/main/java/br/unicamp/cst/util/viewer/ObjectTreeNode.java

```diff
@@ -42,11 +45,19 @@ public DefaultMutableTreeNode addItem(String name, String value, Object ob, int
     
     public DefaultMutableTreeNode addNumber(Number n,String name) {
         String s="";
-        if (n instanceof Long || n instanceof Integer) {
+        if (n instanceof Long) {
```

Comment:
The repeated code is due to a conversion from Number to String, in order to allow visualization. In ObjectTreeNode, it is doing a first initialization. In the MemoryInspector it is doing the refresh of newer values. In order to refactor, it would be possible to create a StringFormatter class, with an static method NumberToString, which could be used in both cases. But because its use is too limited and too particular, maybe it is easier to just repeat the code in both classes. Unless someone thinks it is a good idea to create such a class !!